### DISCUSSION TITLE
Sampling discontinuous likelihood functions

### DISCUSSION THREAD

**Discussion by LegrandNico** [blackjax-devs/blackjax#496] [Q&A] [answered]
What is the current recommended solution to sample log probability functions that are discontinuous? 

I have encountered this problem repeatedly while working on [continuous Hierarchical Gaussian filters](https://ilabcode.github.io/pyhgf/notebooks/2-Continuous_HGF.html#continuous-hgf), and we would like to consider alternatives for this specific problem. 

I found [this reference](https://arxiv.org/abs/1705.08510) that seems to address it but I don't know how good this solution is, and if a Python version exists somewhere.

Sorry if this is a duplicate

**Comment by albcab** [comment, marked as answer]
Discontinuous HMC is an interesting algorithm and it would be great to have it implemented in the blackjax library! You are welcome to open an issue where we can discuss the design of its implementation. 

Currently to sample discrete variables in blackjax you would use [Metropolis within Gibbs](https://blackjax-devs.github.io/blackjax/examples/howto_metropolis_within_gibbs.html) to sample from the discrete conditional on the continuous and vice versa. Not sure how useful this is for discontinuous likelihoods though.

**Reply by rlouf** [reply]
In some cases you can (and should) marginalise over the discrete random variables in your model. Aesara/AePPL plans on having this available as a model transformation, for instance.

In every other cases, the solution @albcab mentioned is the way to go.
